DVCodingStyle.md

```diff
@@ -0,0 +1,864 @@
+The class files must be `` `include``-ed into the package file.
+Directory structure of the UVM agent and UVM environment testbench components
+along with their related files are shown below, with some additional guidance on
+declaring functions from a package context.
+
+
+
+
+### `uvm_agent` directory structure
+
+* All DV code for a particular agent will be placed under a single directory
+`<block>_agent/` separate from the DV code for the corresponding environment.
+* It is highly recommended that all agent directories be placed in a common area
+  to create a repository of UVM agents that allows for more vertical reuse.
+
+```systemverilog
+<block>_agent/
+  <block>_if.sv
+  <block>_item.sv
+  <block>_agent.sv
+  <block>_agent_pkg.sv
+  <block>_agent_cfg.sv
+  <block>_agent_cov.sv
+  <block>_driver.sv
+  <block>_host_driver.sv (Only if block requires a separate driver for host mode)
+  <block>_device_driver.sv (Only if block requires a separate driver for device mode)
+  <block>_monitor.sv
+  <block>_sequencer.sv
+  seq_lib/
+    <block>_base_seq.sv
+```
+
+
+### `uvm_env` directory structure
+* All DV code for the UVM environment, UVM tests, and top level block
+  testbench will be placed in a directory `<dut>/dv/` in three corresponding
+  subdirectories `<dut>/dv/env/, <dut>/dv/tests/, <dut>/dv/tb/`.
+* The `<dut>_env_pkg` should import the `<dut>_agent_pkg`.
+* All UVM virtual sequences for a given DUT will be placed under `env/seq_lib/`,
+  along with a `<dut>_vseq_list.sv` file that `` `include``-es all of the virtual
+  sequence files. This virtual sequence list file must be `` `include``-ed by
+  `env/<dut>_env_pkg.sv`.
+* If applicable, `env/<dut>_env_pkg.sv` must import the `<dut>_agent_pkg`.
+* `tests/<dut>_test_pkg.sv` must import the `<dut>_env_pkg` package.
+* All files `` `include``-ed into a single package must be in the same single directory.
+
+```systemverilog
+<dut>/dv/
+  env/
+    <dut>_env.sv
+    <dut>_env_pkg.sv
+    <dut>_env_cfg.sv
+    <dut>_env_cov.sv
+    <dut>_scoreboard.sv
+    <dut>_virtual_sequencer.sv
+    seq_lib/
+      // These allow basic functionality to be added to the environment
+      <dut>_base_vseq.sv
+      <dut>_sanity_vseq.sv
+  tb/
+    tb.sv
+  tests/
+    <dut>_base_test.sv
+    <dut>_test_pkg.sv
+```
+`<block>_agent_pkg.sv, <dut>_env_pkg.sv, <dut>_test_pkg.sv, and tb/tb.sv`
+  must include these `` `include``-es and imports:
+
+```systemverilog
+`include "dv_macros.svh"
+`include "uvm_macros.svh"
+import uvm_pkg::*;
+```
+
+An example environment level package file for an arbitrary block `newblock` should look similar to this:
+
+```systemverilog
+package newblock_env_pkg;
+  // dep packages
+  import uvm_pkg::*;
+  import newblock_agent_pkg::*;
+
+  // macro includes
+  `include "uvm_macros.svh"
+  `include "dv_macros.svh"
+
+  // define any parameters
+  parameter int MY_BLOCK_PARAMETER = 10;
+
+  // define any local types/functions/tasks
+  typedef enum bit {
+    MyblockBooleanZero,
```

Comment:
What's the relationship between newblock and Myblock, or should that be NewblockBoolean...

Reply:
Good catch, let me change this to NewblockBoolean.